modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinClientCodegen.java

```diff
@@ -167,7 +168,7 @@ public KotlinClientCodegen() {
 
         CliOption collectionType = new CliOption(COLLECTION_TYPE, "Option. Collection type to use");
         Map<String, String> collectionOptions = new HashMap<>();
-        collectionOptions.put(CollectionType.ARRAY.value, "kotlin.Array");
+        collectionOptions.put(CollectionType.ARRAY.value, "kotlin.collections.List");
```

Comment:
@kuFEAR why is this a List instead of an ArrayList?
Because by setting `kotlin.collections.List` here, both `CollectionType.ARRAY` and `CollectionType.LIST` do exactly the same.

Reply:
hm, I'm not noticed this option for switching collections type, I'll revert this now

Reply:
but will be better to set List as default collection type for Kotlin, as I see, we don't generate hashCode&equals in response models

Reply:
Thanks for changing this.